code/datums/uplink/uplink_items.dm

```diff
@@ -116,9 +116,8 @@ datum/uplink_item/dd_SortValue()
 		var/list/L = I
 		if(L.len) I = L[1]
 
-	if(istype(I) && ishuman(user))
-		var/mob/living/carbon/human/A = user
-		A.put_in_any_hand_if_possible(I)
```

Comment:
What have you done with the various put in hand functions?

We had many that governed exactly how it should function, like put in hand or delete

Reply:
It was too weird proc used in 2 places.
```
/mob/proc/put_in_any_hand_if_possible(obj/item/W as obj, del_on_fail = 0, disable_warning = 1, redraw_mob = 1)
	if(equip_to_slot_if_possible(W, slot_l_hand, del_on_fail, disable_warning, redraw_mob))
		return TRUE
	else if(equip_to_slot_if_possible(W, slot_r_hand, del_on_fail, disable_warning, redraw_mob))
		return TRUE
	return FALSE
```
1) Priority - left hand - first, right - second. Active hand has no matter
2) If first `equip_to_slot_if_possible` fails item qdels, but after that other `equip..._if_possible` calls
and so on.

Furthemore, deleting item if you have no free hand but it still use your points - not good practice.
Also, returning value isn't check and deleted item (what still hold in `I`) is passing to caller.

FTW. It's easy, safety and more obvious to use `put_in_hands`